planner/core/planbuilder.go

```diff
@@ -432,6 +432,7 @@ type cteInfo struct {
 	// enterSubquery and recursiveRef are used to check "recursive table must be referenced only once, and not in any subquery".
 	enterSubquery bool
 	recursiveRef  bool
+	limitLP       LogicalPlan
```

Comment:
Do we really need limitLP?

Reply:
limitLP will be tableDual if SQL is `limit 0`

Reply:
ok